services/node_manager/src/test/java/com/futurewei/alcor/nodemanager/controller/NodeControllerTest.java

```diff
@@ -0,0 +1,165 @@
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import java.net.InetAddress;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+public class NodeControllerTest {
+    private static final ObjectMapper om = new ObjectMapper();
+    public NodeInfo testNodeInfo;
+    String strTestNodeId = "";
+
+    @Autowired
+    NodeService service;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private NodeController mockController;
+
+    @Before
+    public void init() {
+        byte[] ip = new byte[]{10, 0, 0, 1};
+        NodeInfo nodeInfo = new NodeInfo("h00", "host0", ip, "AA-BB-CC-DD-EE-00");
+        NodeInfoJson nodeInfoJson = new NodeInfoJson(nodeInfo);
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        try {
+            String json = objectMapper.writeValueAsString(nodeInfoJson);
+            NodeInfo nodeInfo2 = service.createNodeInfo(nodeInfo);
+            strTestNodeId = nodeInfo2.getId();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public String createNodeInfo(NodeInfo nodeInfo) {
+        NodeInfoJson nodeInfoJson = new NodeInfoJson(nodeInfo);
+        ObjectMapper objectMapper = new ObjectMapper();
+        String strNodeId = "";
+        try {
+            String json = objectMapper.writeValueAsString(nodeInfoJson);
+            NodeInfo nodeInfo2 = service.createNodeInfo(nodeInfo);
+            strTestNodeId = nodeInfo2.getId();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return strTestNodeId;
+    }
+
+    @Test
+    public void test_index() throws Exception {
+        this.mockMvc.perform(get("/start.html"))
+                .andDo(print())
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    public void test_getNodeInfoFromUpload() throws Exception {
+        MockMultipartFile file = new MockMultipartFile("file", "@/mnt/dev/alcor/config/machine.json", "application/json", "{\"host_info\":{\"node_id\":\"ephost_0\",\"node_name\":\"ephost_0\",\"local_ip\":\"172.17.0.6\",\"mac_address\":\"02:42:ac:11:00:06\",\"veth\":\"\",\"server_port\":50001}}" .getBytes());
```

Comment:
Could you move the test machine.json to somewhere under /test folder?

Reply:
It is moved.